src/ast.rs

```diff
@@ -2198,7 +2224,9 @@ where
                 write!(ctx, "::")?;
                 member.demangle(ctx, scope)
             }
-        }
+        };
+        ctx.pop_demangle_node();
```

Comment:
If you don't care that early returns (for the error cases) don't run this code I think you should say that explicitly in a comment here.